web/src/main/java/com/bazaarvoice/emodb/web/resources/databus/DatabusEventWriterAdminImpl.java

```diff
@@ -0,0 +1,155 @@
+        }
+        return updateRef;
+    }
+
+    private Map<String, Object> get(String tableName, String key, ReadConsistency consistency) {
+        Table table = _tableDao.get(tableName);
+
+        // Query from the database
+        Record record = _dataReaderDao.read(new Key(table, key), consistency);
+
+        // Resolve the deltas into a single object
+        Resolved resolved = _recordResolver.resolve(record, consistency);
+
+        // Convert to the final JSON format including intrinsic fields
+        return _recordResolver.toContent(resolved, consistency, true);
+    }
+
+    @Override
+    public Iterator<UpdateRef> writeEvents(final String table, final int batchSize, final ReadConsistency consistency, @Nullable final Date date) {
+        Objects.requireNonNull(table);
+        Objects.requireNonNull(consistency);
+        //TODO
+        return Collections.EMPTY_LIST.iterator();
+/*        Collection<String> splits = _dataStore.getSplits(table, batchSize);
+
+        final Iterator<UpdateRef>[] result = new Iterator[] {Iterators.emptyIterator()};
+        Predicate<Map<String, Object>> predicate = getDatePredicate(date);
+        Set<String> tags = new HashSet<>(Arrays.asList("replay"));
+
+        splits.parallelStream().forEach(split -> {
+            Iterator<Map<String, Object>> documents = _dataStore.getSplit(table, split, null, Long.MAX_VALUE, true, consistency);
+            Iterable<Map<String, Object>> iterable = () -> documents;
+            long now = System.currentTimeMillis();
+            long fct = _dataWriterDao.getFullConsistencyTimestamp(_tableDao.get(table));
+
+            List<UpdateRef> updateRefs = StreamSupport.stream(iterable.spliterator(), true)
+                    .filter(predicate)
+                    .map(document -> {
+                        long lastUpdateAt = Intrinsic.getLastUpdateAt(document).getTime();
+                        UUID changeId = getChangeId(lastUpdateAt, fct, now);
+                        return new UpdateRef(table, Intrinsic.getId(document), changeId, tags);
+                    })
+                    .collect(Collectors.toList());
+
+            writeEvents(updateRefs);
+            result[0] = Iterators.concat(result[0], updateRefs.iterator());
+        });
+        return result[0];*/
+    }
+
+    @Override
+    public void writeEvents(final Collection<UpdateRef> refs) {
+        _eventWriterRegistry.getDatabusWriter().writeEvents(refs);
+    }
+
+    private static Predicate<Map<String, Object>> getDatePredicate(Date date) {
+        Predicate<Map<String, Object>> datePredicate = document -> Intrinsic.getFirstUpdateAt(document).compareTo(date) >= 0;
+        Predicate<Map<String, Object>> alwaysTruePredicate = document -> true;
+        return date == null ? alwaysTruePredicate : datePredicate;
+    }
+
+    private static UUID getChangeId(long lastUpdateAt, final UUID lastCompactedMutation, final UUID lastCompactionCutoff, long fct, long now) {
+        //TODO
```

Comment:
I see what you are doing here. Getting lastCompactedMutation and lastCompactionCutoff values and making sure the databus event is not ignored as per the exact condition mentioned in "isChangeDeltaRedundant" method. 

The other approach I was thinking is to use FCT value directly to make it cleaner,  but that may not be accurate as the above values I see are beyond FCT. 
_cutoff = UUID of the oldest consistent delta not deleted by compaction
_lastContentMutation = UUID of the most recent consistent delta which changed the resolved object deleted by compaction.  

Overall, I like using those 2 values here instead of FCT which may not be right way. If so, you can scrap brining in the fct value I think.

Reply:
great, but i think we will still need FCT for rare cases when there are no compactions present for document.